apps/src/lab2/levelEditors/validations/EditValidations.tsx

```diff
@@ -0,0 +1,164 @@
+import Typography from '@cdo/apps/componentLibrary/typography/Typography';
+import {Validation, ConditionType, AppName} from '../../types';
+import {MusicConditions} from '@cdo/apps/music/progress/MusicValidator';
+import {convertOptionalStringToBoolean} from '@cdo/apps/types/utils';
+import {createUuid} from '@cdo/apps/utils';
+import React, {useState} from 'react';
+import moduleStyles from './edit-validations.module.scss';
+import EditValidation from './EditValidation';
+
+/**
+ * This component is used to edit validations for a level. Currently only used
+ * by Lab2 labs.
+ */
+
+const AppConditions: {[key in AppName]?: ConditionType[]} = {
+  music: Object.values(MusicConditions),
+};
+
+interface EditValidationsProps {
+  initialValidations: Validation[];
+  levelName: string;
+  appName: AppName;
+}
+
+/**
+ * Top-level validations editor component.
+ */
+const EditValidations: React.FunctionComponent<EditValidationsProps> = ({
+  initialValidations,
+  levelName,
+  appName,
+}) => {
+  const [validations, setValidations] = useState<Validation[]>(
+    sanitizeValidations(initialValidations, levelName)
+  );
+
+  const conditionTypes = AppConditions[appName];
+
+  if (conditionTypes === undefined || conditionTypes.length === 0) {
+    return (
+      <div>
+        {`No validation conditions available for app type: ${appName}. Contact the engineering team for further details.`}
+      </div>
+    );
+  }
+
+  const onValidationChange = (validation: Validation) => {
+    setValidations(
+      sanitizeValidations(
+        validations.map(v => {
+          if (v.key === validation.key) {
+            return validation;
+          }
+          return v;
+        }),
+        levelName
+      )
+    );
+  };
+
+  const deleteValidation = (key: string) => {
+    setValidations(
+      sanitizeValidations(
+        validations.filter(validation => {
+          return validation.key !== key;
+        }),
+        levelName
+      )
+    );
+  };
+
+  const moveValidation = (key: string, direction: 'up' | 'down') => {
```

Comment:
It might be tough given the size of a validation's UI block, but it's pretty slick when we can drag things to reorder them.  The up/down arrows do seem sufficient though.

Reply:
Agreed! Wanted to just get something working so I went with the arrows, but it would be cool to pull in some React dragging library for this